src/core/lib/channel/status_util.cc

```diff
@@ -109,3 +111,28 @@ bool grpc_status_code_from_int(int status_int, grpc_status_code* status) {
   *status = static_cast<grpc_status_code>(status_int);
   return true;
 }
+
+namespace grpc_core {
+
+absl::Status MaybeRewriteIllegalStatusCode(const absl::Status& status,
```

Comment:
If we changed this signature to:

```
absl::Status MaybeRewriteIllegalStatusCode(absl::Status status, absl::string_view source)
```
we could std::move the status into this function in the majority of (all?) cases, and then use the RVO to make this rewriting essentially free even if there's an error message attached to a valid status.

Reply:
Done.